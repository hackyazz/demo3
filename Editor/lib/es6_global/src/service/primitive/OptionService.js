

import * as Curry from "../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Caml_option from "../../../../../node_modules/bs-platform/lib/es6/caml_option.js";
import * as OptionService$Wonderjs from "../../../../../node_modules/wonder.js/lib/es6_global/src/service/atom/OptionService.js";

function join(x) {
  if (x !== undefined) {
    return Caml_option.valFromOption(x);
  }
  
}

function andThenWithDefault(func, $$default, x) {
  if (x !== undefined) {
    return Curry._1(func, Caml_option.valFromOption(x));
  } else {
    return $$default;
  }
}

function either(someFunc, noneFunc, data, x) {
  if (x !== undefined) {
    return Curry._2(someFunc, data, Caml_option.valFromOption(x));
  } else {
    return Curry._1(noneFunc, data);
  }
}

function eitherWithNoData(someFunc, noneFunc, x) {
  if (x !== undefined) {
    return Curry._1(someFunc, Caml_option.valFromOption(x));
  } else {
    return Curry._1(noneFunc, /* () */0);
  }
}

function handleSomeAndIgnore(func, x) {
  if (x !== undefined) {
    Curry._1(func, Caml_option.valFromOption(x));
    return /* () */0;
  } else {
    return /* () */0;
  }
}

var unsafeGet = OptionService$Wonderjs.unsafeGet;

var unsafeGetJsonSerializedValue = OptionService$Wonderjs.unsafeGetJsonSerializedValue;

var isJsonSerializedValueNone = OptionService$Wonderjs.isJsonSerializedValueNone;

export {
  unsafeGet ,
  unsafeGetJsonSerializedValue ,
  isJsonSerializedValueNone ,
  join ,
  andThenWithDefault ,
  either ,
  eitherWithNoData ,
  handleSomeAndIgnore ,
  
}
/* OptionService-Wonderjs Not a pure module */
