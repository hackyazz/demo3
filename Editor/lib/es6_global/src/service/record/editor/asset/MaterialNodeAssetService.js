

import * as Block from "../../../../../../../node_modules/bs-platform/lib/es6/block.js";
import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Log$WonderLog from "../../../../../../../node_modules/wonder-log/lib/es6_global/src/Log.js";
import * as LogUtils$WonderEditor from "../../../../core/utils/console/LogUtils.js";

function getNodeName(param, getMaterialNameFunc) {
  return Curry._2(getMaterialNameFunc, param[/* materialComponent */1], param[/* type_ */0]);
}

function buildNodeData(type_, materialComponent) {
  return /* record */[
          /* type_ */type_,
          /* materialComponent */materialComponent
        ];
}

function buildNode(nodeId, type_, materialComponent) {
  return /* MaterialNode */Block.__(1, [
            nodeId,
            /* record */[
              /* type_ */type_,
              /* materialComponent */materialComponent
            ]
          ]);
}

function buildNodeByNodeData(nodeId, nodeData) {
  return /* MaterialNode */Block.__(1, [
            nodeId,
            nodeData
          ]);
}

function isMaterialNode(node) {
  if (node.tag === 1) {
    return true;
  } else {
    return false;
  }
}

function getNodeData(node) {
  if (node.tag === 1) {
    return node[1];
  } else {
    return Log$WonderLog.fatal(LogUtils$WonderEditor.buildFatalMessage("should be material node", "", "", ""));
  }
}

export {
  getNodeName ,
  buildNodeData ,
  buildNode ,
  buildNodeByNodeData ,
  isMaterialNode ,
  getNodeData ,
  
}
/* Log-WonderLog Not a pure module */
