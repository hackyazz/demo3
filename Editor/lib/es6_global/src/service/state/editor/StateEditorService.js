

import * as CreateEditorStateDataEditorService$WonderEditor from "../data/CreateEditorStateDataEditorService.js";

function getStateIsDebug(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1];
}

function setStateIsDebug(isDebug) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1] = isDebug;
  return /* () */0;
}

function getIsRun(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2];
}

function setIsRun(isRun) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2] = isRun;
  return /* () */0;
}

function getState(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0];
}

function setState(state) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0] = state;
  return state;
}

export {
  getStateIsDebug ,
  setStateIsDebug ,
  getIsRun ,
  setIsRun ,
  getState ,
  setState ,
  
}
/* CreateEditorStateDataEditorService-WonderEditor Not a pure module */
