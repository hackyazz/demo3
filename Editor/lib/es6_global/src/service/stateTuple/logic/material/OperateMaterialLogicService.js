

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as BasicMaterialEngineService$WonderEditor from "../../../state/engine/BasicMaterialEngineService.js";
import * as LightMaterialEngineService$WonderEditor from "../../../state/engine/LightMaterialEngineService.js";

function getNewMaterilaName(param) {
  return "New Material";
}

function getDefaultName(param) {
  return "NoName Material";
}

function _getName(material, engineState, getMaterialNameFunc) {
  var match = Curry._2(getMaterialNameFunc, material, engineState);
  if (match !== undefined) {
    return match;
  } else {
    return "NoName Material";
  }
}

function getName(material, type_, engineState) {
  if (type_) {
    return _getName(material, engineState, LightMaterialEngineService$WonderEditor.getLightMaterialName);
  } else {
    return _getName(material, engineState, BasicMaterialEngineService$WonderEditor.getBasicMaterialName);
  }
}

function setName(material, type_, name, engineState) {
  if (type_) {
    return LightMaterialEngineService$WonderEditor.setLightMaterialName(material, name, engineState);
  } else {
    return BasicMaterialEngineService$WonderEditor.setBasicMaterialName(material, name, engineState);
  }
}

export {
  getNewMaterilaName ,
  getDefaultName ,
  _getName ,
  getName ,
  setName ,
  
}
/* BasicMaterialEngineService-WonderEditor Not a pure module */
