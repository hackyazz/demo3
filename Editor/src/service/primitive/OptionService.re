open Wonderjs;

let unsafeGet = OptionService.unsafeGet;

let unsafeGetJsonSerializedValue = OptionService.unsafeGetJsonSerializedValue;

let isJsonSerializedValueNone = OptionService.isJsonSerializedValueNone;

let join = x =>
  switch (x) {
  | None => None
  | Some(x) => x
  };

let andThenWithDefault = (func, default, x) =>
  switch (x) {
  | None => default
  | Some(x) => func(x)
  };

let either = (someFunc, noneFunc, data, x) =>
  switch (x) {
  | None => noneFunc(data)
  | Some(x) => someFunc(data, x)
  };

let eitherWithNoData = (someFunc, noneFunc, x) =>
  switch (x) {
  | None => noneFunc()
  | Some(x) => someFunc(x)
  };

let handleSomeAndIgnore = (func, x) =>
  switch (x) {
  | None => ()
  | Some(x) => func(x) |> ignore
  };