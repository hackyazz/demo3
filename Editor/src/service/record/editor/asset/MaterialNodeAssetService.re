open TreeAssetType;

open NodeAssetType;

let getNodeName =
    ({materialComponent, type_}: materialNodeData, getMaterialNameFunc) =>
  getMaterialNameFunc(~material=materialComponent, ~type_);

let buildNodeData = (~type_, ~materialComponent) => {
  type_,
  materialComponent,
};

let buildNode = (~nodeId, ~type_, ~materialComponent) =>
  MaterialNode(nodeId, buildNodeData(~type_, ~materialComponent));

let buildNodeByNodeData = (~nodeId, ~nodeData) =>
  MaterialNode(nodeId, nodeData);

let isMaterialNode = node =>
  switch (node) {
  | MaterialNode(_, _) => true
  | _ => false
  };

let getNodeData = node =>
  switch (node) {
  | MaterialNode(nodeId, nodeData) => nodeData
  | _ =>
    WonderLog.Log.fatal(
      LogUtils.buildFatalMessage(
        ~description={j|should be material node|j},
        ~reason="",
        ~solution={j||j},
        ~params={j||j},
      ),
    )
  };