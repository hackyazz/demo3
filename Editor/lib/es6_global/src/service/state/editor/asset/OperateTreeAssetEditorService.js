

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Js_option from "../../../../../../../node_modules/bs-platform/lib/es6/js_option.js";
import * as NodeAssetService$WonderEditor from "../../../record/editor/asset/NodeAssetService.js";
import * as FolderNodeAssetService$WonderEditor from "../../../record/editor/asset/FolderNodeAssetService.js";
import * as TreeAssetEditorService$WonderEditor from "./TreeAssetEditorService.js";
import * as IterateTreeAssetService$WonderEditor from "../../../record/editor/asset/IterateTreeAssetService.js";
import * as OperateTreeAssetService$WonderEditor from "../../../record/editor/asset/OperateTreeAssetService.js";
import * as MaterialNodeAssetService$WonderEditor from "../../../record/editor/asset/MaterialNodeAssetService.js";
import * as CurrentNodeIdAssetEditorService$WonderEditor from "./CurrentNodeIdAssetEditorService.js";

function _getTree(func, editorState) {
  return Curry._1(func, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
}

function _getTreeWithLabel(func, editorState) {
  return Curry._1(func, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
}

function _getAndSetTree(func, editorState) {
  var __x = Curry._1(func, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
  return TreeAssetEditorService$WonderEditor.setTree(__x, editorState);
}

function findNodeById(targetNodeId, editorState) {
  var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
  return OperateTreeAssetService$WonderEditor.findNodeById(targetNodeId, param);
}

function unsafeFindNodeById(targetNodeId, editorState) {
  var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
  return OperateTreeAssetService$WonderEditor.unsafeFindNodeById(targetNodeId, param);
}

function isNodeExistById(targetNodeId, editorState) {
  var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
  return Js_option.isSome(OperateTreeAssetService$WonderEditor.findNodeById(targetNodeId, param));
}

function findMaterialNode(targetMaterialComponent, targetType_, editorState) {
  var predNodeFunc = function (materialNode) {
    var match = MaterialNodeAssetService$WonderEditor.getNodeData(materialNode);
    if (match[/* materialComponent */1] === targetMaterialComponent) {
      return match[/* type_ */0] === targetType_;
    } else {
      return false;
    }
  };
  var arg = predNodeFunc;
  return (function (param) {
              return IterateTreeAssetService$WonderEditor.findOne(param, undefined, arg, undefined, undefined, /* () */0);
            })(TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
}

function insertNode(targetNodeId, newTreeNode, editorState) {
  return _getAndSetTree((function (param) {
                return OperateTreeAssetService$WonderEditor.insertNode(targetNodeId, newTreeNode, param);
              }), editorState);
}

function setNodeIsShowChildren(targetNodeId, isShowChildren, editorState) {
  return _getAndSetTree((function (param) {
                return OperateTreeAssetService$WonderEditor.setNodeIsShowChildren(targetNodeId, isShowChildren, param);
              }), editorState);
}

function removeNode(targetNode, editorState) {
  var newTree = OperateTreeAssetService$WonderEditor.removeNode(targetNode, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
  return TreeAssetEditorService$WonderEditor.setTree(newTree, editorState);
}

function removeNodeById(targetNodeId, editorState) {
  var match = OperateTreeAssetService$WonderEditor.removeNodeById(targetNodeId, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
  var newTree = match[0];
  return /* tuple */[
          (function (param) {
              return TreeAssetEditorService$WonderEditor.setTree(newTree, param);
            }),
          match[1]
        ];
}

function findNodeParent(targetNode, editorState) {
  return IterateTreeAssetService$WonderEditor.findOne(TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState), undefined, undefined, undefined, (function (node) {
                return Js_option.isSome(FolderNodeAssetService$WonderEditor.findChild(node, targetNode));
              }), /* () */0);
}

function findNodeParentId(targetNode, editorState) {
  return Js_option.map(NodeAssetService$WonderEditor.getNodeId, findNodeParent(targetNode, editorState));
}

function getCurrentNode(editorState) {
  return Js_option.map((function (nodeId) {
                var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
                return OperateTreeAssetService$WonderEditor.unsafeFindNodeById(nodeId, param);
              }), CurrentNodeIdAssetEditorService$WonderEditor.getCurrentNodeId(editorState));
}

function unsafeGetSelectedFolderNodeInAssetTree(editorState) {
  var targetNodeId = TreeAssetEditorService$WonderEditor.getSelectedFolderNodeIdInAssetTree(editorState);
  var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
  return OperateTreeAssetService$WonderEditor.unsafeFindNodeById(targetNodeId, param);
}

export {
  _getTree ,
  _getTreeWithLabel ,
  _getAndSetTree ,
  findNodeById ,
  unsafeFindNodeById ,
  isNodeExistById ,
  findMaterialNode ,
  insertNode ,
  setNodeIsShowChildren ,
  removeNode ,
  removeNodeById ,
  findNodeParent ,
  findNodeParentId ,
  getCurrentNode ,
  unsafeGetSelectedFolderNodeInAssetTree ,
  
}
/* FolderNodeAssetService-WonderEditor Not a pure module */
