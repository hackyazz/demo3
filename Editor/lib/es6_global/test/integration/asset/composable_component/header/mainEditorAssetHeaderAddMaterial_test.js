

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as OptionService$WonderEditor from "../../../../../src/service/primitive/OptionService.js";
import * as ReactTestTool$WonderEditor from "../../../../tool/ReactTestTool.js";
import * as BuildComponentTool$WonderEditor from "../../../../tool/BuildComponentTool.js";
import * as StateEditorService$WonderEditor from "../../../../../src/service/state/editor/StateEditorService.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../tool/MainEditorSceneTool.js";
import * as MainEditorAssetIdTool$WonderEditor from "../../tool/MainEditorAssetIdTool.js";
import * as AssetTreeInspectorTool$WonderEditor from "../../../inspector/composable_component/assetTree_inspector/tool/AssetTreeInspectorTool.js";
import * as LightMaterialToolEngine$WonderEditor from "../../../../tool/engine/LightMaterialToolEngine.js";
import * as MainEditorAssetTreeTool$WonderEditor from "../../tool/MainEditorAssetTreeTool.js";
import * as MaterialNodeAssetService$WonderEditor from "../../../../../src/service/record/editor/asset/MaterialNodeAssetService.js";
import * as OperateMaterialLogicService$WonderEditor from "../../../../../src/service/stateTuple/logic/material/OperateMaterialLogicService.js";
import * as OperateTreeAssetEditorService$WonderEditor from "../../../../../src/service/state/editor/asset/OperateTreeAssetEditorService.js";
import * as MainEditorAssetMaterialNodeTool$WonderEditor from "../../tool/MainEditorAssetMaterialNodeTool.js";
import * as MainEditorAssetHeaderOperateNodeTool$WonderEditor from "../../tool/MainEditorAssetHeaderOperateNodeTool.js";

describe("MainEditorAssetHeader->add material", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        var _prepareAndExecAndGetMaterialNode = function () {
          var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Material */3][/* buildOneMaterialAssetTree */0], /* () */0);
          var addedMaterialNodeId = MainEditorAssetIdTool$WonderEditor.getNewAssetId(undefined, /* () */0);
          MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
          return /* tuple */[
                  assetTreeData,
                  OperateTreeAssetEditorService$WonderEditor.unsafeFindNodeById(addedMaterialNodeId, StateEditorService$WonderEditor.getState(/* () */0))
                ];
        };
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, (function () {
                              return /* () */0;
                            }));
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("if not select specific treeNode", (function () {
                describe("should add material into root treeNode", (function () {
                        Wonder_jest.test("test snapshot", (function () {
                                Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Material */3][/* buildOneMaterialAssetTree */0], /* () */0);
                                MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetComponent(/* () */0));
                              }));
                        return Wonder_jest.test("the added material parent node should be root", (function () {
                                      var match = _prepareAndExecAndGetMaterialNode(/* () */0);
                                      return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](OptionService$WonderEditor.unsafeGet(MainEditorAssetTreeTool$WonderEditor.findNodeParentId(match[1], StateEditorService$WonderEditor.getState(/* () */0)))), Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Material */3][/* getRootNodeId */1], match[0]));
                                    }));
                      }));
                return /* () */0;
              }));
        describe("else", (function () {
                describe("add material into specific treeNode", (function () {
                        return Wonder_jest.test("test snapshot", (function () {
                                      var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* All */6][/* ThreeLayer */0][/* buildFolderAndTextureAndMaterialAssetTree */0], /* () */0);
                                      MainEditorAssetTreeTool$WonderEditor.Select[/* selectFolderNode */1](Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* All */6][/* ThreeLayer */0][/* getSecondLayerFirstFolderNodeId */2], assetTreeData), undefined, /* () */0);
                                      MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetTree(/* () */0));
                                    }));
                      }));
                return /* () */0;
              }));
        Wonder_jest.test("create new material", (function () {
                Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Material */3][/* buildOneMaterialAssetTree */0], /* () */0);
                var addedMaterialNodeId = MainEditorAssetIdTool$WonderEditor.getNewAssetId(undefined, /* () */0);
                var newLightMaterial = LightMaterialToolEngine$WonderEditor.getNewLightMaterial(undefined, /* () */0);
                MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
                var materialComponent = MainEditorAssetMaterialNodeTool$WonderEditor.getMaterialComponent(addedMaterialNodeId, undefined, /* () */0);
                return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](materialComponent), newLightMaterial);
              }));
        Wonder_jest.test("material type should be LightMaterial", (function () {
                var match = _prepareAndExecAndGetMaterialNode(/* () */0);
                return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](MaterialNodeAssetService$WonderEditor.getNodeData(match[1])[/* type_ */0]), /* LightMaterial */1);
              }));
        describe("test name", (function () {
                Wonder_jest.test("test default name", (function () {
                        Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Material */3][/* buildOneMaterialAssetTree */0], /* () */0);
                        MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
                        return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetComponent(/* () */0));
                      }));
                return Wonder_jest.test("remove first material which use default name;\n          add three material;\n\n          the first new one's name should be removed-material's name;\n                  ", (function () {
                              var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Material */3][/* buildOneMaterialAssetTree */0], /* () */0);
                              var nodeId = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Material */3][/* getFirstMaterialNodeId */2], assetTreeData);
                              AssetTreeInspectorTool$WonderEditor.Rename[/* renameAssetMaterialNode */2](undefined, undefined, nodeId, OperateMaterialLogicService$WonderEditor.getDefaultName(/* () */0), /* () */0);
                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.removeMaterialNode(undefined, undefined, nodeId, /* () */0);
                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.addMaterial(undefined, undefined, /* () */0);
                              MainEditorAssetTreeTool$WonderEditor.Select[/* selectFolderNode */1](Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Material */3][/* getRootNodeId */1], assetTreeData), undefined, /* () */0);
                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetChildrenNode(undefined, /* () */0));
                            }));
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
