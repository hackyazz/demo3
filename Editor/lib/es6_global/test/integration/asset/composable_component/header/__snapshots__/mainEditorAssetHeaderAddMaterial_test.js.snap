// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorAssetHeader->add material else add material into specific treeNode test snapshot 1`] = `
<article
  className="wonder-asset-assetTree"
>
  <ul
    className="wonder-tree-node"
  >
    <li>
      <div
        className="item-triangle"
        onMouseDown={[Function]}
      >
        <img
          src="./public/img/down.png"
        />
      </div>
      <img
        className="treeNode-icon"
        src="./public/img/package.png"
      />
      <div
        className="draggable-container"
        draggable={true}
        onClick={[Function]}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        style={Object {}}
      >
        Assets
      </div>
    </li>
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
          onMouseDown={[Function]}
        >
          <img
            src="./public/img/down.png"
          />
        </div>
        <img
          className="treeNode-icon"
          src="./public/img/package.png"
        />
        <div
          className="draggable-container select-active"
          draggable={true}
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          NoName Folder
        </div>
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
          onMouseDown={[Function]}
        >
          <img
            src="./public/img/down.png"
          />
        </div>
        <img
          className="treeNode-icon"
          src="./public/img/package.png"
        />
        <div
          className="draggable-container"
          draggable={true}
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          NoName Folder 1
        </div>
      </li>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            NoName Folder
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            NoName Folder 1
          </div>
        </li>
      </ul>
    </ul>
  </ul>
</article>
`;

exports[`MainEditorAssetHeader->add material if not select specific treeNode should add material into root treeNode test snapshot 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={false}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
            onMouseDown={[Function]}
          >
            <img
              src="./public/img/down.png"
            />
          </div>
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container select-not-active"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Assets
          </div>
        </li>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  >
    <article
      className="wonder-asset-fileBox "
      onClick={[Function]}
    >
      <div
        className="box-image"
      >
        <img
          onDragStart={[Function]}
          src="./public/img/mat.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          material1
        </span>
      </div>
    </article>
    <article
      className="wonder-asset-fileBox "
      onClick={[Function]}
    >
      <div
        className="box-image"
      >
        <img
          onDragStart={[Function]}
          src="./public/img/mat.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          New Material
        </span>
      </div>
    </article>
  </article>
</article>
`;

exports[`MainEditorAssetHeader->add material test name remove first material which use default name;
          add three material;

          the first new one's name should be removed-material's name;
                   1`] = `
<article
  className="wonder-asset-assetChildren"
>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/mat.png"
      />
    </div>
    <div
      className="item-text "
    >
      <span>
        New Material
      </span>
    </div>
  </article>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/mat.png"
      />
    </div>
    <div
      className="item-text "
    >
      <span>
        New Material 1
      </span>
    </div>
  </article>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/mat.png"
      />
    </div>
    <div
      className="item-text "
    >
      <span>
        New Material 2
      </span>
    </div>
  </article>
</article>
`;

exports[`MainEditorAssetHeader->add material test name test default name 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={false}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
            onMouseDown={[Function]}
          >
            <img
              src="./public/img/down.png"
            />
          </div>
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container select-not-active"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Assets
          </div>
        </li>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  >
    <article
      className="wonder-asset-fileBox "
      onClick={[Function]}
    >
      <div
        className="box-image"
      >
        <img
          onDragStart={[Function]}
          src="./public/img/mat.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          material1
        </span>
      </div>
    </article>
    <article
      className="wonder-asset-fileBox "
      onClick={[Function]}
    >
      <div
        className="box-image"
      >
        <img
          onDragStart={[Function]}
          src="./public/img/mat.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          New Material
        </span>
      </div>
    </article>
  </article>
</article>
`;
